### ISSUE TITLE
New install issue with Scala kernel & Plot()

### ISSUE THREAD

**Issue by jaety** [twosigma/beakerx#5883]
Very excited by the talk yesterday at JupyterCon. Looks like great stuff. Trying to get started I immediately ran into the following error though:

I followed the initial instructions to install the beakerx extensions and then launched the Scala kernel. I tried to simply execute the single cell

`new Plot()`

but that comes back with the error 

`Error rendering Jupyter widget. Widget not found: {"model_id":"db6f2b63-19b1-4af3-9511-214986356ee3"}`

I'm a little stuck as to next debugging steps to take. I did check to ensure I had the 6.0.0 version of ipywidgets installed, as that seemed from another issue to be a hard dependency. 
Thanks.

**Comment by scottdraves** [comment]
thanks for the report, I'll take a look when I get back to my desk.

**Comment by jpallas** [comment]
This isn't specific to one kernel.  I got similar messages with Groovy and Python.  I used miniconda on macOS to create an install environment with `conda create -n beakerx-test python=3.5 jupyter`.

**Comment by scottdraves** [comment]
are you building master or using "pip install"?
are there any errors on the JS console?
what is the output of "jupyter nbextension list"?

**Comment by jpallas** [comment]
I did a `pip install`.  I haven't tried a master build in a completely fresh environment.

There is something on the JS console: `Wrong widget protocol version: received protocol version '', but was expecting major version '2'`.  Before trying to run "new Plot()" there were three messages about failing to load resources for `.js.map` files.  My brief contact with JS tooling makes me believe those are only used in debugging?  (Parenthetically, is Safari deliberately making it hard to copy from the JS console?  Sheesh!)

`jupyter nbextension list` output:
```
Known nbextensions:
  config dir: /Users/joe/miniconda2/envs/beakerx-test/etc/jupyter/nbconfig
    notebook section
      jupyter-js-widgets/extension  enabled 
      - Validating: OK
      beakerx/extension  enabled 
      - Validating: OK
```

(update) Here's all of the JS console output:
```
[Warning] actions (2) (main.min.js, line 33264)
"jupyter-notebook:find-and-replace""does not exist, still binding it in case it will be defined later..."
[Log] load_extensions – Arguments (main.min.js, line 12429)
Arguments0: "jupyter-js-widgets/extension"1: "beakerx/extension"callee: TypeError: 'arguments', 'callee', and 'caller' cannot be accessed in this context. TypeError: 'arguments', 'callee', and 'caller' cannot be accessed in this context.length: 2Symbol(Symbol.iterator): function()Object Prototype
[Log] Loading extension: jupyter-js-widgets/extension (main.min.js, line 12406)
[Log] Loading extension: beakerx/extension (main.min.js, line 12406)
[Log] LOAD – Object (extension.js, line 51042)
Object
[Log] Session: kernel_created (e5cb620f-73d7-4ada-822a-63999bee91c8) (main.min.js, line 24777)
[Log] Starting WebSockets: – "ws://localhost:8890/api/kernels/e2fb97e3-8086-45e4-b0ab-941a938dfd5c" (main.min.js, line 23933)
[Log] Kernel: kernel_connected (e2fb97e3-8086-45e4-b0ab-941a938dfd5c) (main.min.js, line 23577)
[Log] Kernel: kernel_ready (e2fb97e3-8086-45e4-b0ab-941a938dfd5c) (main.min.js, line 23577)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (index.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (preact.min.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (preact-compat.min.js.map, line 0)
[Error] Wrong widget protocol version: received protocol version '', but was expecting major version '2'
	handle_comm_open (extension.js:62600)
	(anonymous function)
	(anonymous function) (main.min.js:23222)
	promiseReactionJob
[Error] Couldn't process kernel message – "Wrong widget protocol version: received protocol version '', but was expecting major version '2'"
	(anonymous function) (main.min.js:24481)
	promiseReactionJob
[Warning] mimebundle missing metadata – {output_type: "display_data", transient: undefined, data: {application/vnd.jupyter.widget-view+json: {model_id: "ae93546e-4258-44b7-a81d-7dcad06d1658"}}, …} (main.min.js, line 21182)
{output_type: "display_data", transient: undefined, data: {application/vnd.jupyter.widget-view+json: {model_id: "ae93546e-4258-44b7-a81d-7dcad06d1658"}}, metadata: undefined}Objectdata: {application/vnd.jupyter.widget-view+json: {model_id: "ae93546e-4258-44b7-a81d-7dcad06d1658"}}Objectmetadata: {}Objectoutput_type: "display_data"transient: undefinedObject Prototype
```

**Comment by jaety** [comment]
I built using `pip install`. outputs below from nbextension and console.log look at least superficially same as jpallas saw. 

`jupyter nbextension list`

```
Known nbextensions:
  config dir: //anaconda/envs/beakerx2/etc/jupyter/nbconfig
    notebook section
      beakerx/extension  enabled 
      - Validating: OK
      jupyter-js-widgets/extension  enabled 
      - Validating: OK
```
JS Console Output. 

```
manager-base.js:134 Wrong widget protocol version: received protocol version '', but was expecting major version '2'
ManagerBase.handle_comm_open @ manager-base.js:134
(anonymous) @ comm.js:89
Promise resolved (async)
CommManager.comm_open @ comm.js:85
i @ jquery.min.js:4
Kernel._handle_iopub_message @ kernel.js:1223
Kernel._finish_ws_message @ kernel.js:1015
(anonymous) @ kernel.js:1006
Promise resolved (async)
Kernel._handle_ws_message @ kernel.js:1006
i @ jquery.min.js:4
kernel.js:1007 Couldn't process kernel message Wrong widget protocol version: received protocol version '', but was expecting major version '2'
(anonymous) @ kernel.js:1007
Promise rejected (async)
Kernel._handle_ws_message @ kernel.js:1007
i @ jquery.min.js:4
kernel.js:1007 Couldn't process kernel message Wrong widget protocol version: received protocol version '', but was expecting major version '2'
(anonymous) @ kernel.js:1007
Promise rejected (async)
Kernel._handle_ws_message @ kernel.js:1007
i @ jquery.min.js:4
kernel.js:1007 Couldn't process kernel message Wrong widget protocol version: received protocol version '', but was expecting major version '2'
(anonymous) @ kernel.js:1007
Promise rejected (async)
Kernel._handle_ws_message @ kernel.js:1007
i @ jquery.min.js:4
outputarea.js:277 mimebundle missing metadata {output_type: "display_data", transient: undefined, data: {…}, metadata: undefined}
OutputArea.validate_mimebundle @ outputarea.js:277
OutputArea.append_output @ outputarea.js:321
OutputArea.handle_output @ outputarea.js:243
output @ codecell.js:365
Kernel._handle_output_message @ kernel.js:1196
i @ jquery.min.js:4
Kernel._handle_iopub_message @ kernel.js:1223
Kernel._finish_ws_message @ kernel.js:1015
(anonymous) @ kernel.js:1006
Promise resolved (async)
Kernel._handle_ws_message @ kernel.js:1006
i @ jquery.min.js:4
```

**Comment by scottdraves** [comment]
thanks for the detailed report.  I was able to reproduce this and it seems like the release of widgets v7 has caused some confusion between the packages.  I was able to fix it with the following:
```
conda install widgetsnbextension=2.0.0
```
because it turned out I had 3.0.1 and that is for 7.0.0 of the ipywidgets package, but i had 6.0.0.

seems like this is an upstream bug, maybe a problem with skew in conda packaging.
i reported it here: https://github.com/jupyter-widgets/ipywidgets/issues/1678

once we get beakerx working with widgets 7 (very soon i think) then we'll make a new release and i think this problem will go away.  i'll update our readme in the meantime.
